src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs

```diff
@@ -735,6 +738,36 @@ class DebugItem
             public Task<byte[][]> Data { get; set; }
         }
 
+        public IEnumerable<SourceFile> Add(SessionId sessionId, byte[] assembly_data, byte[] pdb_data)
+        {
+            var resolver = new DefaultAssemblyResolver();
```

Comment:
Is there any reason not to make the resolver a readonly member?  I think it might avoid some extra searching overhead.

Reply:
I don't see any reason not too. I can make this change.

Reply:
The lifetime of the DebugStore is a bit different than the runtime assemblies so the reload resolution might be slightly different?

Reply:
From reading the code, I think it should be fine. Initializing the resolver once will allow both `Add` and `Load` to share the same cache, which I believe is what we want here.